apps/dotcom-asset-upload/src/index.ts

```diff
@@ -10,145 +11,27 @@ interface Env {
 	UPLOADS: R2Bucket
 }
 
-function parseRange(
-	encoded: string | null
-): undefined | { offset: number; end: number; length: number } {
-	if (encoded === null) {
-		return
-	}
-
-	const parts = (encoded.split('bytes=')[1]?.split('-') ?? []).filter(Boolean)
-	if (parts.length !== 2) {
-		console.error('Not supported to skip specifying the beginning/ending byte at this time')
-		return
-	}
-
-	return {
-		offset: Number(parts[0]),
-		end: Number(parts[1]),
-		length: Number(parts[1]) + 1 - Number(parts[0]),
-	}
-}
-
-function objectNotFound(objectName: string): Response {
-	return new Response(`<html><body>R2 object "<b>${objectName}</b>" not found</body></html>`, {
-		status: 404,
-		headers: {
-			'content-type': 'text/html; charset=UTF-8',
-		},
-	})
-}
-
-const CACHE_CONTROL_SETTING = 's-maxage=604800'
-
 const router = Router()
 
 router
 	.all('*', preflight)
-	.get('/uploads/:objectName', async (request: Request, env: Env, ctx: ExecutionContext) => {
-		const url = new URL(request.url)
-
-		const range = parseRange(request.headers.get('range'))
-
-		// NOTE: caching will only work when this is deployed to
-		// a custom domain, not a workers.dev domain. It's a no-op
-		// otherwise.
-
-		// Construct the cache key from the cache URL
-		const cacheKey = new Request(url.toString(), request)
-		const cache = caches.default as Cache
-
-		// Check whether the value is already available in the cache
-		// if not, you will need to fetch it from R2, and store it in the cache
-		// for future access
-		let cachedResponse
-		if (!range) {
-			cachedResponse = await cache.match(cacheKey)
-
-			if (cachedResponse) {
-				return cachedResponse
-			}
-		}
-
-		const ifNoneMatch = request.headers.get('if-none-match')
-		let hs = request.headers
-		if (ifNoneMatch?.startsWith('W/')) {
-			hs = new Headers(request.headers)
-			hs.set('if-none-match', ifNoneMatch.slice(2))
-		}
```

Comment:
this was working around a bug https://github.com/cloudflare/workers-sdk/issues/3089 i'm not sure if they've fixed it so might be worth either checking or just keeping the code in case

Reply:
oh thanks for pointing that out! I'll take a look